notify/sns/sns.go

```diff
@@ -0,0 +1,190 @@
+	}
+	return &Notifier{
+		conf:    c,
+		tmpl:    t,
+		logger:  l,
+		client:  client,
+		retrier: &notify.Retrier{},
+	}, nil
+}
+
+func (n *Notifier) Notify(ctx context.Context, alert ...*types.Alert) (bool, error) {
+	var (
+		err   error
+		data                           = notify.GetTemplateData(ctx, n.tmpl, alert, n.logger)
+		tmpl                           = notify.TmplText(n.tmpl, data, &err)
+		creds *credentials.Credentials = nil
+	)
+	if n.conf.Sigv4.AccessKey != "" && n.conf.Sigv4.SecretKey != "" {
+		creds = credentials.NewStaticCredentials(n.conf.Sigv4.AccessKey, string(n.conf.Sigv4.SecretKey), "")
+	}
+
+	attributes := make(map[string]*sns.MessageAttributeValue, len(n.conf.Attributes))
+	if len(n.conf.Attributes) > 0 {
+		for k, v := range n.conf.Attributes {
+			attributes[tmpl(k)] = &sns.MessageAttributeValue{DataType: aws.String("String"), StringValue: aws.String(tmpl(v))}
+		}
+
+	}
+
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config: aws.Config{
+			Region:      aws.String(tmpl(n.conf.Sigv4.Region)),
+			Credentials: creds,
+			Endpoint:    aws.String(tmpl(n.conf.APIUrl)),
+		},
+		Profile: n.conf.Sigv4.Profile,
+	})
+
+	if _, err := sess.Config.Credentials.Get(); err != nil {
+		return false, fmt.Errorf("could not get SigV4 credentials: %w", err)
+	}
+
+	if n.conf.Sigv4.RoleARN != "" {
+		sess.Config.Credentials = stscreds.NewCredentials(sess, n.conf.Sigv4.RoleARN)
+	}
+
+	// Max message size for a message in a SNS publish request is 256KB, except for SMS messages where the limit is 1600 characters/runes.
+	messageSize := 256 * 1024
+	client := sns.New(sess, &aws.Config{Credentials: creds})
+	publishInput := &sns.PublishInput{}
+
+	if n.conf.TopicARN != "" {
+		publishInput.SetTopicArn(tmpl(n.conf.TopicARN))
+
+		if n.isFifo == nil {
+			checkFifo, err := checkTopicFifoAttribute(client, n.conf.TopicARN)
+			if err != nil {
+				if e, ok := err.(awserr.RequestFailure); ok {
+					return n.retrier.Check(e.StatusCode(), strings.NewReader(e.Message()))
+				} else {
+					return true, err
+				}
+			}
+			n.isFifo = &checkFifo
+		}
+		if *n.isFifo {
+			// Deduplication key and Message Group ID are only added if it's a FIFO SNS Topic.
+			key, err := notify.ExtractGroupKey(ctx)
+			if err != nil {
+				return false, err
+			}
+			publishInput.SetMessageDeduplicationId(key.Hash())
+			publishInput.SetMessageGroupId(key.Hash())
+		}
+	}
+	if n.conf.PhoneNumber != "" {
+		publishInput.SetPhoneNumber(tmpl(n.conf.PhoneNumber))
+		// If we have an SMS message, we need to truncate to 1600 characters/runes.
+		messageSize = 1600
+	}
+	if n.conf.TargetARN != "" {
+		publishInput.SetTargetArn(tmpl(n.conf.TargetARN))
+
+	}
+
+	messageToSend, isTrunc, err := validateAndTruncateMessage(tmpl(n.conf.Message), messageSize)
+	if err != nil {
+		return false, err
+	}
+	if isTrunc {
+		attributes[tmpl("truncated")] = &sns.MessageAttributeValue{DataType: aws.String("String"), StringValue: aws.String(tmpl("true"))}
```

Comment:
Not sure I understand why you need to call `tmpl()` here.
```suggestion
		attributes["truncated"] = &sns.MessageAttributeValue{DataType: aws.String("String"), StringValue: aws.String("true")}
```